x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/NativeAutodetectProcessFactory.java

```diff
@@ -95,14 +101,19 @@ private void createNativeProcess(Job job, AutodetectParams autodetectParams, Pro
             if (autodetectParams.quantiles() != null) {
                 autodetectBuilder.quantiles(autodetectParams.quantiles());
             }
-
-            autodetectBuilder.build();
+            autodetectBuilder.build(maxAnomalyRecords);
```

Comment:
As we are creating a new `AutodetectBuilder` each time, I think we can pass it the settings in each constructor. We currently pass it the stored settings that could be out of date. But we can pass it instead `clusterService.settings()`. I am not sure what the difference between `clusterService.settings()` and `clusterService.clusterSettings()` is, but if we figure that out I'm sure we can make it work. This would mean we don't need to have a listener in `AutodetectProcessManager` for the setting plus we can keep the `build()` method without any arguments.